core/src/main/java/org/infinispan/interceptors/impl/CallInterceptor.java

```diff
@@ -604,16 +598,7 @@ public Object visitRemoveExpiredCommand(InvocationContext ctx, RemoveExpiredComm
 
    @Override
    public Object visitSizeCommand(InvocationContext ctx, SizeCommand command) throws Throwable {
-      ClusterPublisherManager managerToUse;
-      if (command.hasAnyFlag(FlagBitSets.CACHE_MODE_LOCAL)) {
-         managerToUse = localClusterPublisherManager;
-      } else {
-         managerToUse = clusterPublisherManager;
-      }
-
-      return asyncValue(managerToUse.keyReduction(false, null, null, ctx,
-            command.getFlagsBitSet(), DeliveryGuarantee.EXACTLY_ONCE, PublisherReducers.count(),
-            PublisherReducers.add()));
+      return sizeHandler.size(ctx, command);
```

Comment:
I was thinking we would use the interceptors to instead handle if the cache was distributed for example. This way the `BaseDistributionInterceptor` would handle the SizeCommand in `visitSizeCommand` and the `CallInterceptor` would handle for local caches or when CACHE_MODE_LOCAL flag was set.

Reply:
The other option, which is more aligned with what you have here is instead to use the ClusterPublisherManager directly and add a method to it.  This way we wouldn't duplicate all the code in your `FallbackSizeCommandHandler` class.

Reply:
Went with the approach to create a method in the `ClusterPublisherManager`.